### PR TITLE
feat(user-auth): add changeUsername Mutation

### PR DESCRIPTION
**Pull request by khanjasir90**
## #73 

Signed-off-by: khanjasir90 <khanmohdjasir@gmail.com
Description: Added mutation for changeUsername

### CODE REVIEW

**Comment by khanjasir90** [comment]
https://www.loom.com/share/8596b9f6453a4661b6ecd1a13dfe9b66

File: src/resolvers/userResolver.js [hunk 1]
```diff
@@ -134,6 +134,62 @@ const userResolver = {
       );
     },
 
+    changeUsername: async(_, { userInfo }, { req }) => {
+      const { oldUsername, newUsername } = userInfo;
```
Comment (d-e-v-esh) [inline src/resolvers/userResolver.js]: I do not think we need to extract the `oldUsername` from the `userInfo`. That would require the user to enter their current username by themselves. We can extract the `oldUsername` from the `req` context. Check `getCurrentUser` query to see how that can be done.

File: src/resolvers/userResolver.js [hunk 2]
```diff
@@ -134,6 +134,62 @@ const userResolver = {
...
+
+      if(oldUsername === newUsername) {
+        return {
+          errors: [
+            {
+              field: "username",
+              message: "Old and New Username are same",
```
Comment (d-e-v-esh) [inline src/resolvers/userResolver.js]: I think we should keep all the error messages in lowercase.

**Review by d-e-v-esh — CHANGES_REQUESTED** [review]
This is good. Just few small corrections and it'll be good to go 👍

**Comment by d-e-v-esh** [comment]
Did you run this updated code?

**Comment by khanjasir90** [comment]
actually there are still some changes that are needed to be made and by mistake I have made a commit! I'll do that and commit again.

**Comment by khanjasir90** [comment]
@d-e-v-esh I have made the necessary changes and it's working on my end too.

**Comment by d-e-v-esh** [comment]
You're on the right track. But I think you are getting confused about how to use the id from `req`. I wrote a little bit about it in my GSoC blog. Take a look at it. Maybe it can help. 👍
https://gist.github.com/d-e-v-esh/cd150019b3bb1188e13eb0df9b210ef9#difference-between-_id-and-id

**Comment by khanjasir90** [comment]
> You're on the right track. But I think you are getting confused about how to use the id from `req`. I wrote a little bit about it in my GSoC blog. Take a look at it. Maybe it can help. 👍 https://gist.github.com/d-e-v-esh/cd150019b3bb1188e13eb0df9b210ef9#difference-between-_id-and-id

@d-e-v-esh thanks for the resource! it really helped me :) and I have also made the necessary changes. Thank you

**Comment by d-e-v-esh** [comment]
Yes, I think it looks good. Can you please clean up all the commits and put them into a single commit. Also clean up the console log and unused import, then we can merge it 👍

File: src/resolvers/userResolver.js [hunk 3]
```diff
@@ -177,7 +178,7 @@ const userResolver = {
             errors: [
               {
                 field: "username",
-                message: "Username already exist's",
+                message: "username already exist's",
```
Comment (d-e-v-esh) [inline src/resolvers/userResolver.js]: `username already exists`

File: src/resolvers/userResolver.js [hunk 4]
```diff
@@ -154,7 +155,7 @@ const userResolver = {
           errors: [
             {
               field: "username",
-              message: "Username does'nt exist's",
+              message: "username does'nt exist's",
```
Comment (d-e-v-esh) [inline src/resolvers/userResolver.js]: `username does not exist`

**Review by d-e-v-esh — CHANGES_REQUESTED** [review]

**Comment by khanjasir90** [comment]
@d-e-v-esh yes would do the needful changes and will make a final commit!

**Pull request closed without merging**